src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -228,6 +201,38 @@ ILearningTransportTransaction GetTransaction()
             return new DirectoryBasedTransaction(messagePumpBasePath, PendingDirName, CommittedDirName, Guid.NewGuid().ToString());
         }
 
+        async Task ProcessFileAndComplete(ILearningTransportTransaction transaction, string filePath, string messageId)
+        {
+            try
+            {
+                await ProcessFile(transaction, messageId)
+                    .ConfigureAwait(false);
+            }
+            finally
+            {
+                if (log.IsDebugEnabled)
+                {
+                    log.Debug($"Completing processing for {filePath}({transaction.FileToProcess}).");
```

Comment:
Should we log with a higher severity to make sure that users reports issues to us?

Reply:
@andreasohlund This was a pre-existing log message, and this isn't one that indicates a problem. Are you thinking about a different message?

Also, this PR introduces several other debug-level messages. Are you thinking we'd want to revise all of them?

Reply:
Sorry for the confusion, I meant in general. Will talk to @danielmarbach and PR against develop if needed so no need to hold up this PR